### PR TITLE
 Add gradlePluginPortal() repository metadata source change in 5.0 to upgrading doc

### PR DESCRIPTION
**Pull request by eskatos**
Change in https://github.com/gradle/gradle/pull/6774 and https://github.com/gradle/gradle/pull/7324
Also see https://github.com/gradle/gradle/issues/7584

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/upgrading_version_4.adoc [hunk 1]
```diff
@@ -177,11 +177,12 @@ Miscellaneous::
  * The internal `org.gradle.util` package is no longer imported by default.
 +
 Ideally you shouldn't use classes from this package, but, as a quick fix, you can add explicit imports to your build scripts for those classes.
+ * The `gradlePluginPortal()` repository no more infers artifact metadata from artifact files and now <<#rel5.0:gradle_plugin_portal_metadata,only uses Maven POMs>>.
```
Comment (oehme) [inline subprojects/docs/src/docs/userguide/upgrading_version_4.adoc]: I think people won't know what that means. How about:

"The `gradlePluginPortal()` repository no longer looks for JARs without a POM by default"

File: subprojects/docs/src/docs/userguide/upgrading_version_4.adoc [hunk 2]
```diff
@@ -494,6 +495,17 @@ Since Java 11 no longer contains the `java.xml.bind` module, Gradle now bundles
 
 Please remove the `--add-modules java.xml.bind` option from `org.gradle.jvmargs`, if set.
 
+[[rel5.0:gradle_plugin_portal_metadata]]
+=== [5.0] `gradlePluginPortal()` repository now fetches artifact metadata from Maven POMs only
+
+The `gradlePluginPortal()` repository is now stricter when resolving artifact metadata, it doesn't infer the metadata from the sole presence of artifact files such as JARs. Only Maven POMs are now considered.
```
Comment (oehme) [inline subprojects/docs/src/docs/userguide/upgrading_version_4.adoc]: Same as above, I think we should shorten and clarify this.

File: subprojects/docs/src/docs/userguide/upgrading_version_4.adoc [hunk 3]
```diff
@@ -494,6 +495,17 @@ Since Java 11 no longer contains the `java.xml.bind` module, Gradle now bundles
...
+
+This repository is used by default when resolving plugins from the `plugins {}` block.
+
+With this new behavior, if a plugin or a transitive dependency of a plugin found in the `gradlePluginPortal()` repository has no Maven POM it will fail to resolve.
+
+Artifacts published to a Maven repository without a POM should be fixed. If you encounter such artifacts, please ask the plugin or library author to publish a new version with proper metadata.
```
Comment (oehme) [inline subprojects/docs/src/docs/userguide/upgrading_version_4.adoc]: There actually is a workaround by adjusting the repository in the pluginManagement block. May be worth pointing out for people who are stuck with a bad plugin.

**Comment by eskatos** [comment]
@oehme thanks! it should be clearer now

**Pull request merged**